packages/cli/src/scaling/scaling.service.ts

```diff
@@ -210,21 +196,34 @@ export class ScalingService {
 				return;
 			}
 
-			if (
-				this.instanceType === 'worker' &&
-				error.message.includes('job stalled more than maxStalledCount')
-			) {
+			throw error;
+		});
+
+		if (this.instanceType === 'main') {
+			this.registerMainListeners();
+		} else if (this.instanceType === 'worker') {
+			this.registerWorkerListeners();
+		}
```

Comment:
Nice, this makes it much easier to understand 👌

Reply:
Once we improve the [duplicate reconnection logic](https://n8nio.slack.com/archives/C069HS026UF/p1723618011118889) (likely moving it to `RedisClientService`) this will be even cleaner :)